### PR TITLE
Create get_update fn to get one update status

### PR DESCRIPTION
**Pull request by icyJoseph**
Hi,

Having fun using this SDK, but I reckon a function to get the status of one single update is missing.

When I read through https://docs.meilisearch.com/reference/api/updates.html#get-an-update-status, I see for example:

```javascript
 client.index('movies').getUpdateStatus(1)
```

Where the argument of `getUpdateStatus`, I guess, is the update id based on what the cURL version does:

```bash
curl \
  -X GET 'http://localhost:7700/indexes/movies/updates/1'
```

Which mirrors the MeiliSearch `/indexes/:index_uid/updates/:updateId` endpoint.

The rather unlucky thing is that to do the same with the Rust SDK it is not sufficient, not possible, to use the update id.

```rust
 let status: Status = progress.get_status().await.unwrap();
```

The above requires access to the `progress` instance, which is not possible in my case. Regardless of any specific use case, _**it is my opinion**_ that for completeness sake, the `get_update` function, defined in this PR, is needed to complete the SDK. Of course this can be challenged, or perhaps there's a reason to why such function does not exist.

Thanks for the SDK! Great Job!

### CODE REVIEW

File: src/progress.rs
Comment (Mubelotix) [inline src/progress.rs:31]:
```suggestion
```

File: src/indexes.rs
Comment (Mubelotix) [inline src/indexes.rs:820]:
```suggestion
        let progress = index.delete_all_documents().await.unwrap();
```
In this test, we just need to get a `Progress`. `Client.delete_all_documents()` is the most compact way to do this. We can also remove the `Movie` struct.

File: src/indexes.rs
Comment (Mubelotix) [inline src/indexes.rs:791]:
```suggestion
```
See next comment

File: src/indexes.rs
Comment (Mubelotix) [inline src/indexes.rs:776]:
```suggestion
    use crate::{client::*, progress::UpdateStatus};
```
See next comment

File: src/indexes.rs
Comment (Mubelotix) [inline src/indexes.rs:657]: I fear that some users will misunderstand this example. They might think that this is the only way to get a `Progress` and they might use this instead of `progress.get_status()`. Could you make this clearer? By telling that they should use `progress.get_status()` in most cases and explaining when this is intended to be used.

**Review by Mubelotix — CHANGES_REQUESTED** [review]

File: src/indexes.rs
Comment (icyJoseph) [reply src/indexes.rs]: I did a couple of changes to try and tackle this.

First in this very function description, I added a sentence that describes the normal course of action, that is, getting the status directly from the progress object.

Second, in the code example, I obtain the status in both ways, from the progress object, and by calling `get_update` on the index with the `update_id`.

I also modified (and fixed after my last commit) the `.code-samples.meilisearch.yml` file, by expanding the `get_update_md` section (formerly `get_update_1`), with code for both use cases.

I am not so good with copy or technical writing, so I was forced to introduce the wording **"progress object"** which might not be ideal.

What do you think? Does it look good now?

Comment (Mubelotix) [reply src/indexes.rs]: Great! I like this documentation.

> I also modified (and fixed after my last commit) the .code-samples.meilisearch.yml file, by expanding the get_update_md section (formerly get_update_1), with code for both use cases.

That would be perfect if it worked but I am not sure that the documentation scraper is powerful enough. Will this work @curquiza ? Is the format of the documentation automatically detected from the field name suffix or is it hardcoded ? 

> I am not so good with copy or technical writing, so I was forced to introduce the wording "progress object" which might not be ideal.

This is indeed not ideal. I would rather say **"`Progress` struct"**. But the second occurrence of "object" is fine in my opinion.

File: .code-samples.meilisearch.yaml
Comment (curquiza) [inline .code-samples.meilisearch.yaml:71]:
```suggestion
get_update_1: |-
  // You can get the status of a `Progress` instance:
  let status: Status = progress.get_status().await.unwrap();

  // Or you can use index to get an update status using its `update_id`:
  let status: Status = index.get_update(1).await.unwrap();
```

File: src/indexes.rs
Comment (curquiza) [reply src/indexes.rs]: Hello! It will indeed not work in the docs, but you can change it like this: https://github.com/meilisearch/meilisearch-rust/pull/133/files#r638115933

**Comment by curquiza** [comment]
bors try

**Comment by icyJoseph** [comment]
What happened to the `wasm-build`? is that done through another action?

**Comment by curquiza** [comment]
> What happened to the wasm-build? is that done through another action?

I removed this test from the settings, my bad 🙂 #138 is not merged yet

**Review by curquiza — APPROVED** [review]
I approve, but I wait for the @Mubelotix approval before any merge 🙂
Thanks again for this PR @icyJoseph

**Review by Mubelotix — APPROVED** [review]
I approve too!

**Comment by curquiza** [comment]
Awesome!!
Thanks to both of you!

bors merge

**Pull request merged**
